### PR TITLE
[WIP] move to poetry

### PR DESCRIPTION
**Pull request by maxispeicher**
*Description of changes:*
Moving from requirements files to poetry for managing dependencies.

For publishing credentials need to be configured according to https://python-poetry.org/docs/repositories/#configuring-credentials.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: b5aa6f96a59431099006337a7b5d34d80112ac4a
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=26103a44-bed1-4a7e-a348-799d3a6d4b43%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 404a3627566523851c911f79647323b4dd472a72
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=50aadb61-58d1-4db0-bad4-8e3fae43761c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: efccd25d0d29e72edc74ebec3edf6447359410ad
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=15473f7e-c0b7-42dc-9b8f-0982b380196c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 6adc6129c7799b8efd0e05fb21cb3f5c29d38d7b
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0d9ed05b-97ff-45e8-8496-4136e05017c4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: .github/workflows/minimal-tests.yml [hunk 1]
```diff
@@ -29,7 +29,9 @@ jobs:
       - name: Install Requirements
         run: |
           python -m pip install --upgrade pip
-          pip install -U -r requirements-dev.txt
+          python -m pip install poetry==1.1.7
```
Comment (jaidisido) [inline .github/workflows/minimal-tests.yml:32]: 1. It looks like installing via pip is discouraged by [poetry](https://python-poetry.org/docs/#installing-with-pip) or is that not important?
2. Any reason for freezing to 1.1.7?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -135,15 +135,15 @@ or
...
+``poetry install --extras "sqlserver"``
 
 * Go to the ``test_infra`` directory
 
 ``cd test_infra``
 
 * Install CDK dependencies:
 
-``pip install -r requirements.txt``
+``poetry install``
```
Comment (jaidisido) [inline CONTRIBUTING.md:146]: Should it be `poetry install --no-dev` perhaps?

**Review by jaidisido — COMMENTED** [review]
Thanks for this left a couple of comments.

One additional one is that the current [docs/environment.yml](https://github.com/awslabs/aws-data-wrangler/blob/main/docs/environment.yml#L14) has -e installation flag which expects a requirements.txt file that would not exist. Wonder if read the docs supports poetry in fact...

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 3b000d649aa26476d7600df749cdc3a27c9e16a0
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d1d78c10-b4ac-4578-afe2-7f585e5bf8c5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: .github/workflows/minimal-tests.yml (hunk 1 above)
Comment (maxispeicher) [reply .github/workflows/minimal-tests.yml:32]: 1. From my experience, installing `poetry` via the curl command has its benefits if you install it on your local system but in CI/CD it does not make a huge difference. Only point might be if at some point `poetry` and `awswrangler` depend on the package but with different versions.
2. I've just pinned it to the latest version, as in the past an update introduced some issues but as it's only in CI/CD we could also unpin it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (maxispeicher) [reply CONTRIBUTING.md:146]: Currently the `dev` dependencies section is empty so it should do exactly the same.

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: fc2c83049692cbced4de214d70d8a09578a7c253
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a6715973-7406-4c1e-89ec-c0075d65a490%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 886cf24c905d9cb173c9f8c6a8af646686aa262c
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=80e897e8-98e5-47bf-b91d-2c1b17ed385a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 3e5b90a6fd83aed1c75fa2eb96e568150fc344dc
* Result: FAILED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=37f22988-746d-41d7-a3d2-9cf21a90c1ca%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jaidisido** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-pDO66x4b9gEu
* Commit ID: 3e5b90a6fd83aed1c75fa2eb96e568150fc344dc
* Result: SUCCEEDED
* [Build Logs](https://8z22lm66x0.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b3befed5-2e55-4e5d-8779-42b169ef3986%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**